src/app/leaves/components/LeaveActionRender.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useMemo, useState } from "react";
+import ActionsMenu from "@/common/ActionMenu/ActionsMenu";
+import {
+  Info as InfoIcon,
+  Trash2 as TrashIcon,
+  Edit as EditIcon,
+  CheckCircle as CheckCircleIcon,
+  AlertCircle as AlertCircleIcon,
+} from "lucide-react";
+import { MemberRole } from "@/lib/constants";
+import { LeaveRequestsTypes, LeavesRequestStatus } from "@/types";
+import { LeaveRequestDetails } from "./LeaveRequestDetails";
+
+const loggedUser = {
+  name: "John Smith",
+  role: "member",
+};
+
+interface Props {
+  leaveRequestDetails: LeaveRequestsTypes;
+}
+
+const LeaveTableActionRender = ({ leaveRequestDetails }: Props) => {
+  const [isOpenViewModal, setIsOpenViewModal] = useState<Boolean>(false);
+
+  const handleViewDetails = () => {
+    setIsOpenViewModal(!isOpenViewModal);
+  };
+
+  const handleEditInfo = () => {
+    console.log("Editing info...");
+  };
+
+  const handleDeleteRequest = () => {
+    console.log("Deleting request...");
+  };
+
+  const handleApprovedRequest = () => {
+    console.log("Approved Request...");
+  };
+
+  const handleRejectedRequest = () => {
+    console.log("Rejected Request...");
+  };
+
+  const baseActions = [
+    {
+      title: "Edit",
+      id: 2,
+      onClick: handleEditInfo,
+      icon: <EditIcon size={16} />,
+    },
+    {
+      title: "Delete",
+      id: 3,
+      onClick: handleDeleteRequest,
+      icon: <TrashIcon size={16} className="stroke-status-inactive" />,
+    },
+  ];
+
+  const viewInfo = [
+    {
+      title: "View Details",
+      id: 1,
+      onClick: handleViewDetails,
+      icon: <InfoIcon size={16} />,
+    },
+  ];
+
+  const statusActions = [
+    {
+      title: "Approve",
```

Comment:
can we use enum here ?

Reply:
No, we don't need to make them enums because these titles are not used in multiple places.